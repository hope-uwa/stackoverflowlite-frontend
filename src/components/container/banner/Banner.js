/* eslint-disable class-methods-use-this */
import React, { Component } from 'react';

class Banner extends Component {
  render() {
    return (
            <div className="row bg-cream align-left pt1 pl2 pb1 pls1 bb-grey">
                <div className="col9">
                        <h1 id="thread-title" className="">Top Questions</h1>
                </div>
                <div className="col3">
                    <div className="row pt1 pb1 ">
                            <button id="askQuestion" className="btn btn-success btn-md btn-shadow">
                                Ask a Question
                            </button>
                    </div>           
                </div>
                
            </div>
    );
  }
}
export default Banner;