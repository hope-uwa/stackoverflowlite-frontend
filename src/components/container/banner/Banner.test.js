import React from 'react';
import { shallow } from 'enzyme';
import Banner from './Banner';


describe('display banner', () => {
  const setup = () => {
    const article = {
      article: {
        featuredImg: 'xyz.jpg',
        title: 'this is the title',
        description: 'this is the description',
        content: 'It is very clear to me',
        updatedAt: '2018-12-07T10:01:09.143Z',
        likes: 5,
        User: {
          username: 'noldor',
          avatarUrl: 'noldor.jpg',
        },
      },
      read: () => {},
    };
    return shallow(<Banner />);
  };
  it('should show page is loading', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
  it('render banner component', () => {
    const wrapper = setup();

    expect(wrapper.find('#askQuestion').text()).toEqual('Ask a Question');
  });
});
