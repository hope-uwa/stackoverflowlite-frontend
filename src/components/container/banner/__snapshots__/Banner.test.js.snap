// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`display banner should show page is loading 1`] = `
<div
  className="row bg-cream align-left pt1 pl2 pb1 pls1 bb-grey"
>
  <div
    className="col9"
  >
    <h1
      className=""
      id="thread-title"
    />
  </div>
  <div
    className="col3"
  >
    <div
      className="row pt1 pb1 "
    />
  </div>
</div>
`;
